ts/screens/wallet/WalletHomeScreen.tsx

```diff
@@ -297,14 +304,25 @@ class WalletHomeScreen extends React.Component<Props, never> {
 
   public render(): React.ReactNode {
     const { potWallets, potTransactions } = this.props;
+
     const wallets = pot.getOrElse(potWallets, []);
+
+    const hasNotSupportedWallets = wallets
```

Comment:
With this check _hasNotSupportedWallets_ could be undefined (when wallets.length === 0)
You could try to apply the opposite logic:
hasNotSupportedWallets: when there are some wallets (> 0) and no one of them is a credit card (which is the only supported)
```javascript
 const hasNotSupportedWallets =
      wallets.length > 0 &&
      wallets.filter(wallet => wallet.type === TypeEnum.CREDIT_CARD).length ===
        0;
```

Reply:
Ok, I updated the control referring to CREDIT_CARD type. I merged the 2 approaches because .find.length gives an error due to the possibly undefined outcome of .find

Reply:
> Ok, the control is updated referring to CREDIT_CARD type. I merged the 2 approaches because .find.length gives an error due to the possibly undefined outcome of .find

Are you sure you need **find** anymore?

```javascript
 const hasNotSupportedWallets =
      wallets.length > 0 &&
      wallets.filter(wallet => wallet.type === TypeEnum.CREDIT_CARD).length ===
        0;
```
Maybe I'm losing something but **filter** is done only when wallets.length > 0 and filters always returns an array (empty if no matching items are been found)

Reply:
Your're right, I read find instead of filter. Condition updated within https://github.com/teamdigitale/io-app/pull/1212/commits/a68edaac50b5b5c9869199b8e17ef2fa9b3f84e9